File: app/src/main/java/com/example/jvmori/myweatherapp/application/BaseApplication.java
package com.example.jvmori.myweatherapp.application;

import com.example.jvmori.myweatherapp.di.component.DaggerAppComponent;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

public class BaseApplication extends DaggerApplication {
    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder().application(this).build();
    }
}

File: app/src/main/java/com/example/jvmori/myweatherapp/application/BaseApplication.kt
package com.example.jvmori.myweatherapp.application

import com.example.jvmori.myweatherapp.di_.component.DaggerAppComponent
import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

class BaseApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.builder().application(this).build()
    }
}